Src/IronPython/Compiler/Ast/TryStatement.cs

```diff
@@ -160,24 +166,35 @@ public class TryStatement : Statement {
                         )
                     );
             } else {
-                result = body;
+                // If the body throws an exception and there's no catch block,
+                // we should still update the current exception
+                result = AstUtils.Try(
+                    body
+                ).Catch(exception,
+                    Ast.Call(AstMethods.SetCurrentException, Parent.LocalContext, exception),
+                    Ast.Rethrow()
+                );
             }
-            
+
             return Ast.Block(
-                GetVariables(lineUpdated, runElse), 
+                GetVariables(lineUpdated, runElse, previousExceptionContext),
+                Ast.Assign(previousExceptionContext, Ast.Call(AstMethods.SaveCurrentException)),
```

Comment:
Something seems off here, not quite sure what exactly. It looks like you're doing the assignment twice (once in `else if (@catch != null)` and another time here)?

Reply:
Looks like I forgot to remove one of the `SaveCurrentException`s when I moved the line down here (it was initially called in both parts of the if/else statement).